controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/SystemUpgrader.java

```diff
@@ -67,6 +62,20 @@ private void deploy(List<SystemApplication> applications, Version target) {
         }
     }
 
+    /** @return true if all applications have converged to the target version in the zone */
+    private boolean deployInZone(ZoneId zone, List<SystemApplication> applications, Version target) {
+        boolean converged = true;
+        for (SystemApplication application : applications) {
+            boolean dependenciesConverged = application.dependencies().stream()
```

Comment:
I suggest extracting a method for this. You can overload `convergedOn(Version, List<SystemApplication>, Zone)` so the next block becomes: `if (convergedOn(version, application.dependencies(), zone))`

Reply:
Fixed.